### PR TITLE
pdal: allow static builds + add several options + unvendor boost & eigen + fix direct dependencies

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
Some informations on this PR:

- pkgconfig file is lowercase: `pdal.pc`
- `zlib` is a direct dependency of `pdal`. `libcurl` is a direct dependency of `arbiter` (vendored for the moment). It was working previously because those libs were dragged indirectly (it could have failed with others transitive options).
- `eigen` and `boost` are unvendored. For boost, it requires to rename the custom boost namespace used in a source file of pdal. pdal depends on boost filesystem library only.
- `with_unwind` option was not robust (could be found even if False).
- add options `with_zlib`, `with_lzma`, `with_lazperf`. `with_lazperf` can't be enable for the moment (missing recipe in CCI).
- bump dependencies. Drawback: since `gdal` 3.2.1 raises if Visual Studio < 2019 (due to a CI issue in CCI), pdal binaries for Visual Studio 2015 & 2017 won't be built anymore.

File: recipes/pdal/all/conanfile.py [hunk 1]
```diff
@@ -55,30 +52,63 @@ def config_options(self):
...
             raise ConanInvalidConfiguration ("This compiler version is unsupported")
 
+    def requirements(self):
+        # TODO package improvements:
+        # - switch from vendored arbiter (not in CCI). disabled openssl and curl are deps of arbiter
+        # - switch from vendor/nlohmann to nlohmann_json (in CCI)
+        self.requires("boost/1.76.0")
+        self.requires("eigen/3.3.9")
+        self.requires("gdal/3.2.1")
+        self.requires("libcurl/7.75.0") # mandotory dependency of arbiter (to remove if arbiter is unvendored)
```
Comment (SpaceIm) [inline recipes/pdal/all/conanfile.py:69]:
```suggestion
        self.requires("libcurl/7.75.0") # mandatory dependency of arbiter (to remove if arbiter is unvendored)
```

**Review by prince-chrismc — COMMENTED** [review]
Could some of the replace in files be moved to patches.... otherwise I will approve

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Comment by sxsong1207** [comment]
> * remove hardcoded RPATH in shared libs.

Hi there, 

I don't think removing RPATH is a good idea. I have suffered a situation with PDAL.

I linked with my target, which has RUNPATH point to PDAL lib correctly. The system can find pdal_base.so from RUNPATH, however, it cannot find the dependencies 'pdal_utils.so' required by pdal_base.  I checked the conanfile.py of PDAL 2.3.0 and think this is because RPATH is removed.

Is there any specific reason why RPATH has to be removed?

Thank you,
Shaun

**Comment by SpaceIm** [comment]
I would say that binaries installed by conan-center are relocatable beasts, they don't have rpath, not even relative. If you want to run an executable linked to these shared libraries, `virtualrunenv` generator  (or the more recent but still experimental `VirtualRunEnv`) should be used to inject path of all shared libs to `LD_LIBRARY_PATH`.

/cc @SSE4 @jgsogo @uilianries

**Comment by sxsong1207** [comment]
@SpaceIm Agree, my current solution is changing the RUNPATH to RPATH by passing target_link_options(NAME PRIVATE "-Wl,--disable-new-dtags"). So at least it can find dependencies from those pathes.

**Pull request merged**